### DISCUSSION TITLE
`getStaticPaths` with `fallback: false` behavior not regenerating JSON files if missing

### DISCUSSION THREAD

**Discussion by pavellazar** [vercel/next.js#61092] [Help]
### Summary

When using `getStaticPaths` and `getStaticProps` with `fallback: false` it will add the static paths under `routes` section of the `prerender-manifest.json`. 

If after the build, the `JSON` file associated with the static path gets deleted, accessing the page will result into a 404 instead of re-generating the `JSON` file.

Based on this documentation: https://nextjs.org/docs/pages/api-reference/functions/get-static-paths#fallback-false, my understanding is that the resources for the page listed in `getStaticPaths` should be regenerated. 

Is this a bug or some misunderstanding from my part?

### Additional information

```js
export async function getStaticPaths() {
  return {
    paths: ['/page1', '/page2'],
    fallback: false
  }
}

function delay(time: any) {
  return new Promise(resolve => setTimeout(resolve, time));
} 

export async function getStaticProps() {
  await delay(5000);
  return {
    props: {
      generatedAt: new Date().toString(),
    },
    revalidate: 60
  }
}
```

`prerender-manifest.json`
```
{
  "version": 3,
  "routes": {
    "/static-page1": {
      "initialRevalidateSeconds": 60,
      "srcRoute": null,
      "dataRoute": "/_next/data/fWPHDMwfrWSKVw399JZPO/static-page1.json"
    },
    "/page1": {
      "initialRevalidateSeconds": 60,
      "srcRoute": "/[...slug]",
      "dataRoute": "/_next/data/fWPHDMwfrWSKVw399JZPO/page1.json"
    },
    "/page2": {
      "initialRevalidateSeconds": 60,
      "srcRoute": "/[...slug]",
      "dataRoute": "/_next/data/fWPHDMwfrWSKVw399JZPO/page2.json"
    },
    "/": {
      "initialRevalidateSeconds": 60,
      "srcRoute": null,
      "dataRoute": "/_next/data/fWPHDMwfrWSKVw399JZPO/index.json"
    }
  },
  "dynamicRoutes": {
    "/[...slug]": {
      "routeRegex": "^/(.+?)(?:/)?$",
      "dataRoute": "/_next/data/fWPHDMwfrWSKVw399JZPO/[...slug].json",
      "fallback": false,
      "dataRouteRegex": "^/_next/data/fWPHDMwfrWSKVw399JZPO/(.+?)\\.json$"
    }
  },
  "notFoundRoutes": [],
  "preview": {
   
  }
}
```

In the above configuration, if `"dataRoute": "/_next/data/fWPHDMwfrWSKVw399JZPO/page1.json"` gets deleted it will not be regenerated even though it has a `"initialRevalidateSeconds": 60,` set
```


### Example

_No response_

**Comment by icyJoseph** [comment]
Hi,

If I understand correctly, you say you delete the manifest file, and then expect GetStaticPaths with fallback false, to regenerate at runtime?

The problem is that when you set fallback to false, the information about which pages are allowed is created at build time. IIRC, GetStaticPaths doesn't get copied into the runtime. During runtime with fallback true, or blocking, the server first checks if there is a page ready to serve, or a 404 or a redirect, and serves that. It might also revalidate that.

However if, fallback false was used, then the runtime server is very strict, and it uses static information about which pages are available.

**Comment by pavellazar** [comment]
Not the `prerender-manifest.json`, the JSON associated with a page, e.g., for page1, if `/_next/data/fWPHDMwfrWSKVw399JZPO/page1.json` gets deleted, it will not be regenerated
